### PR TITLE
Bump ktlint to version 0.46.1 

### LINKED ISSUE #5022: Update ktlint to 0.46.1
## Expected Behavior
Dependencies are kept up to date.

## Current Behavior
We're on 0.45.2.

## Context
Latest update has breaking API changes which detekt must adapt to: https://github.com/pinterest/ktlint/releases/tag/0.46.0

### PR DESCRIPTION
**Pull request by kvn-stgl**
See https://github.com/pinterest/ktlint/releases/tag/0.46.0 and https://github.com/pinterest/ktlint/releases/tag/0.46.1 for full release notes.

This PR is part of the ktlint upgrade to version 0.46. See #5040

It includes:
- Upgrade to ktlint version 0.46.11
- Changed the rules from experimental ruleset to the standard ruleset which changed with the new version (see https://github.com/pinterest/ktlint/pull/1481). These are also tagged with @ActiveByDefault v1.22
- Added the NoBlankLinesInChainedMethodCallsRule (because it's not a new rule, just an old rule which is split up into two rules)

Closes #5022

### CODE REVIEW

File: scripts/github-milestone-report.main.kts [hunk 1]
```diff
@@ -7,7 +7,8 @@
  * You need kotlin 1.3.70+ installed on your machine
  */
 
-@file:Suppress("detekt.CommentSpacing") // for the exec line
+// for the exec line
+@file:Suppress("detekt.CommentSpacing")
```
Comment (BraisGabin) [inline scripts/github-milestone-report.main.kts:11]: 🤔 this change seems a bit odd, isn't it?

**Comment by BraisGabin** [comment]
Thank you for split it :) this was way way easier to review :)

File: scripts/github-milestone-report.main.kts (hunk 1 above)
Comment (kvn-stgl) [reply scripts/github-milestone-report.main.kts]: Yes, it seems odd. I'm not sure whether we have a false positive here or not. The problem is the `AnnotationOnSeparateLineRule` rule from detekt: `/home/runner/work/detekt/detekt/scripts/github-milestone-report.main.kts:10:1: Annotations with parameters should all be placed on separate lines prior to the annotated construct [AnnotationOnSeparateLine]`

File: detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/wrappers/SpacingBetweenDeclarationsWithAnnotations.kt [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
 import io.gitlab.arturbosch.detekt.api.Config
+import io.gitlab.arturbosch.detekt.api.internal.ActiveByDefault
 import io.gitlab.arturbosch.detekt.api.internal.AutoCorrectable
 import io.gitlab.arturbosch.detekt.formatting.FormattingRule
 
 /**
  * See [ktlint-website](https://ktlint.github.io#rule-spacing) for documentation.
  */
 @AutoCorrectable(since = "1.10.0")
+@ActiveByDefault(since = "1.21.0")
```
[the remark this answers is not in the document]
Comment (kvn-stgl) [reply detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/wrappers/SpacingBetweenDeclarationsWithAnnotations.kt]: Seems like I forgot this line in the cherry-pick. Thx!

**Review by BraisGabin — APPROVED** [review]

File: scripts/github-milestone-report.main.kts (hunk 1 above)
Comment (kvn-stgl) [reply scripts/github-milestone-report.main.kts:11]: I wrote a bug ticket at the ktlint project to fix this issue: https://github.com/pinterest/ktlint/issues/1539

**Pull request merged**